torchvision/prototype/transforms/_auto_augment.py

```diff
@@ -526,3 +526,302 @@ def forward(self, *inputs: Any) -> Any:
+                    angle=0.0,
+                    translate=[int(magnitude), 0],
+                    scale=1.0,
+                    shear=[0.0, 0.0],
+                )
+            elif transform_id == "TranslateY":
+                bboxes = bboxes.affine(
+                    angle=0.0,
+                    translate=[0, int(magnitude)],
+                    scale=1.0,
+                    shear=[0.0, 0.0],
+                )
+            elif transform_id == "ShearX":
+                bboxes = bboxes.affine(
+                    angle=0.0,
+                    translate=[0, 0],
+                    scale=1.0,
+                    shear=[math.degrees(math.atan(magnitude)), 0.0],
+                    center=[0, 0],
+                )
+            elif transform_id == "ShearY":
+                bboxes = bboxes.affine(
+                    angle=0.0,
+                    translate=[0, 0],
+                    scale=1.0,
+                    shear=[0.0, math.degrees(math.atan(magnitude))],
+                    center=[0, 0],
+                )
+        elif transform_id == "BBox_Cutout":
+            raise NotImplementedError()
+        else:
+            image = self._apply_image_or_video_transform(image, transform_id, magnitude, interpolation, fill)
+
+        return image, bboxes
+
+
+class AutoAugmentDetection(_AutoAugmentDetectionBase):
+    _AUGMENTATION_SPACE = {
+        "AutoContrast": (lambda num_bins, height, width: None, False),
+        "Equalize": (lambda num_bins, height, width: None, False),
+        "Posterize": (
+            lambda num_bins, height, width: (8 - (torch.arange(num_bins) / ((num_bins - 1) / 6))).round().int(),
+            False,
+        ),  # TODO: tf repo is (0 -> 4) while this seems to be (8 -> 4)
+        "Solarize": (
+            lambda num_bins, height, width: torch.linspace(255.0, 0.0, num_bins),
+            False,
+        ),  # TODO: check order and 255.0 rather than 256.0
+        "SolarizeAdd": (
+            lambda num_bins, height, width: torch.linspace(0.0, 110.0, num_bins),
+            False,
+        ),  # TODO: check 110.0
+        "Color": (lambda num_bins, height, width: torch.linspace(0.0, 0.9, num_bins), True),
+        "Contrast": (lambda num_bins, height, width: torch.linspace(0.0, 0.9, num_bins), True),
+        "Brightness": (lambda num_bins, height, width: torch.linspace(0.0, 0.9, num_bins), True),
+        "Sharpness": (lambda num_bins, height, width: torch.linspace(0.0, 0.9, num_bins), True),
+        "Cutout": (lambda num_bins, height, width: torch.linspace(0.0, 100.0, num_bins), False),  # TODO: int()?
+        "BBox_Cutout": (lambda num_bins, height, width: torch.linspace(0.0, 0.75, num_bins), False),
+        "Rotate_BBox": (lambda num_bins, height, width: torch.linspace(0.0, 30.0, num_bins), True),
+        "TranslateX_BBox": (
+            lambda num_bins, height, width: torch.linspace(0.0, 250.0, num_bins),
+            True,
+        ),  # TODO: check 250.0
```

Comment:
@ain-soph Here I do not know what should we follow. Paper says [-150, 150] but the code uses [250](https://github.com/tensorflow/tpu/blob/01b67067be64987e9519b35342ba91d3179daeed/models/official/detection/utils/autoaugment_utils.py#L1621)

@datumbox for image-only AA I see the same question:

- https://github.com/pytorch/vision/blob/135a0f9ea9841b6324b4fe8974e2543cbb95709a/torchvision/prototype/transforms/_auto_augment.py#L169-L171
- https://github.com/tensorflow/tpu/blob/b24729de804fdb751b06467d3dce0637fa652060/models/official/efficientnet/autoaugment.py#L667

So, it was decided to implement paper params instead of official code ones ?

Another question, with current linspace definition I have an impression that we pick a wrong value for TranslateX. Is it OK or is it a bug ? For example, policy defines `("TranslateX_BBox", 0.6, 4)` -> `magnitude_idx=4`. 

```
a = torch.linspace(0.0, 250.0, 10, dtype=torch.long)
a[4] 
> 111.11111450195312
```
but in TF it is 100:
```
level = 4
_MAX_LEVEL=10
translate_const = 250
level = (level/_MAX_LEVEL) * float(translate_const)
level
> 100
```
https://github.com/tensorflow/tpu/blob/01b67067be64987e9519b35342ba91d3179daeed/models/official/detection/utils/autoaugment_utils.py#L1420